core/controllers/profile.py

```diff
@@ -214,21 +214,14 @@ def put(self):
                     % feconf.MAX_BIO_LENGTH_IN_CHARS)
 
             user_services.update_user_bio(self.user_id, data)
-        elif update_type == 'subject_interests':
```

Comment:
What about the if above. Also, you should maybe refactor the `update_*` functions so that we only call one that updates all the values.

Reply:
done